core/templates/services/html-length.service.ts

```diff
@@ -71,70 +78,67 @@ export class HtmlLengthService {
+    this.calculateBaselineLength(sanitizedHtml, CALCULATION_TYPE_CHARACTER);
     // Identify custom tags using regex on the original HTML string.
-    const customTagRegex = /<oppia-noninteractive-(?:math|image)[^>]*>/g;
-    const customTags = htmlString.match(customTagRegex);
+    const customTags = htmlString.match(CUSTOM_TAG_REGEX);
 
     if (customTags) {
       for (const customTag of customTags) {
         totalCharacters += this.getWeightForNonTextNodes(
-          customTag, 'character');
+          customTag, CALCULATION_TYPE_CHARACTER);
       }
     }
     return totalCharacters;
   }
 
   private calculateBaselineLength(
-      sanitizedHtml: string, wordOrCharacter: string): number {
+      sanitizedHtml: string, calculationType: CalculationType): number {
     let domparser = new DOMParser();
     let dom = domparser.parseFromString(sanitizedHtml, 'text/html');
     let totalWeight = 0;
     for (let tag of Array.from(dom.body.children)) {
       const ltag = tag.tagName.toLowerCase();
       if (this.textTags.includes(ltag)) {
         totalWeight +=
-        this.getWeightForTextNodes(tag as HTMLElement, wordOrCharacter);
+        this.getWeightForTextNodes(tag as HTMLElement, calculationType);
       }
     }
     return totalWeight;
   }
 
   private getWeightForTextNodes(
-      textNode: HTMLElement, wordOrCharacter: string): number {
+      textNode: HTMLElement, calculationType: CalculationType): number {
     const textContent = textNode.textContent || '';
 
-    if (wordOrCharacter === 'word') {
-      const words = textContent.trim().split(' ');
-      const wordCount = words.length;
-      return wordCount;
+    let textContentTrimmed = textContent.trim();
+    let textContentCount: number = 0;
+    if (calculationType === CALCULATION_TYPE_WORD) {
+      const words = textContentTrimmed.split(' ');
+      textContentCount = words.length;
+    } else if (calculationType === CALCULATION_TYPE_CHARACTER) {
+      const characters = textContentTrimmed.split('\n').join('').split('');
+      textContentCount = characters.length;
     }
-    const characters = textContent.trim().split('\n').join('').split('');
-    let characterCount = characters.length;
-    return characterCount;
+    return textContentCount;
   }
 
   private getWeightForNonTextNodes(
-      nonTextNode: string, wordOrCharacter: string): number {
+      nonTextNode: string, calculationType: CalculationType): number {
     if (nonTextNode.includes('oppia-noninteractive-math')) {
       return 1;
     }
     const altTextMatch = nonTextNode
       .match(/alt-with-value="&amp;quot;([^&]*)&amp;quot;"/);
 
-    if (wordOrCharacter === 'word') {
-      let words = [];
-      if (altTextMatch && altTextMatch[1]) {
-        const altText = altTextMatch[1];
-        words = altText.trim().split(' ');
-      }
-      return words.length + 2; // +2 as a bonus for images with text.
-    }
-    let chars = [];
+    let text = [];
```

Comment:
An array is not a "text".

Instead, store as a variable the alt text string (which can be the empty string if there is no alt text). You already have a function for computing lengths of regular strings; you can probably use it.

Reply:
@seanlip We can reuse the function if we include space characters also in the alternate text.!! So, please can you confirm that!?

Reply:
It's fine for the alt text string to contain spaces.

Reply:
done!